community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/notification/InternalNotification.scala

```diff
@@ -31,3 +31,5 @@ case class CartesianProductNotification(position: InputPosition) extends Interna
 case class LengthOnNonPathNotification(position: InputPosition) extends InternalNotification
 
 case object LegacyPlannerNotification extends InternalNotification
+
+case object CostForUpdatesNotification extends InternalNotification
```

Comment:
The thinking at the time was that this was a temporary feature that would 'soon' be removed, and so making it specific to the current case made sense. For the runtime we went the generic route, because we have no idea how long the transition from interpreted->compiled will take and the border will constantly move.